awx/main/notifications/webhook_backend.py

```diff
@@ -32,8 +38,15 @@ def send_messages(self, messages):
         sent_messages = 0
         if 'User-Agent' not in self.headers:
             self.headers['User-Agent'] = "Tower {}".format(get_awx_version())
+        if self.http_method.lower() not in ['put','post']:
```

Comment:
Due to this change here[1] this is not necessary anymore . We know it will only be `PUT` or `POST`

[1] https://github.com/ansible/awx/pull/4124/files#diff-a81324c523b41de7296fdd5ff9063d10R4240

Reply:
I disagree; I'd prefer to program defensively here in case the code ever changes (perhaps in the future, the validation code in the serializer code stops working properly, or perhaps we grow some additional ORM code that sets this method without properly validating it.

Reply:
Fair. The thing that bothers me a bit is the DRY principles not being followed here.

```
if field == "http_method" and field_val.lower() not in ['put', 'post']
```

and 

```
if self.http_method.lower() not in ['put','post']:
```

Could we have `['put', 'post']` set somewhere and have those statement becomes something like `if self.http_method.lower() not in SUPPORTED_HTTP_METHOD` like statement ?

This approach - I think - would satisfy both point of views

Reply:
I'm a big fan of DRY as long as it actually saves you something 😄.  This is a one line if statement that's succinct, easy to read, and not likely to change drastically.

Reply:
I'm a bit late to this conversation, but I wanted to show why I left that code in the `webhook_backend.py` file.  IF somehow there is a nonsense HTTP method that gets past the UI _and_ the serializer validator (admittedly this is a very unlikely scenario), without that code this is the error that shows up in the UI:

![NoCatchInBackend](https://user-images.githubusercontent.com/28930622/61716614-8bacc500-ad2d-11e9-80cb-6e92e8dffe58.png)

This doesn't really explain what exactly went wrong.  The code that I left in `webhook_backend.py` causes this error message to show up instead, which explains what exactly the user can fix:

![ErrorCaughtInBackend](https://user-images.githubusercontent.com/28930622/61716658-9bc4a480-ad2d-11e9-8581-6bec2b68826a.png)